File: legacy-java/domain/src/main/java/com/streamfirst/iceberg/hybrid/domain/CommitId.java
package com.streamfirst.iceberg.hybrid.domain;

import java.util.Objects;

/**
 * Unique identifier for a table metadata commit in the global catalog. Each commit represents a
 * specific version of table metadata (schema, partitioning, etc.). Commit IDs are globally unique
 * and monotonically increasing per table.
 *
 * @param value the commit identifier (typically a UUID or timestamp-based ID)
 */
public record CommitId(String value) {
  public CommitId {
    Objects.requireNonNull(value, "Commit ID cannot be null");
    if (value.trim().isEmpty()) {
      throw new IllegalArgumentException("Commit ID cannot be empty");
    }
  }

  @Override
  public String toString() {
    return value;
  }
}

File: modules/domain/src/main/scala/com/streamfirst/iceberg/hybrid/domain/CommitId.scala
package com.streamfirst.iceberg.hybrid.domain

import java.util.UUID

/** Strong type for commit identifiers in Apache Iceberg. Represents a specific version of table
  * metadata and schema changes.
  */
opaque type CommitId = String

object CommitId:
  def apply(value: String): CommitId =
    require(value.trim.nonEmpty, "Commit ID cannot be null or empty")
    value

  def generate(): CommitId = s"commit-${UUID.randomUUID()}"

  extension (commitId: CommitId)
    def value: String = commitId
    def asString: String = commitId

  given Conversion[CommitId, String] = _.value
